File: leetcode.com/1221_split_a_string_in_balanced_strings/solution.c
int balancedStringSplit(char *s)
{
    int total = 0;
    int r = 0;
    int l = 0;

    for (int i = 0 ; i < strlen(s) ; i++) {
        if (s[i] == 'R') r++; else l++;

        if (r == l) {
            r = 0;
            l = 0;
            total++;
        }
    }

    return total;
}

File: leetcode.com/1221_split_a_string_in_balanced_strings/solution.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        total = rc = lc = 0

        for ch in s:
            if ch == 'R':
                rc += 1
            else:
                lc += 1

            if rc == lc:
                total += 1
                rc = 0
                lc = 0

        return total
